keras_cv/layers/preprocessing/maybe_apply_test.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from absl.testing import parameterized
+
+from keras_cv.layers.preprocessing.maybe_apply import MaybeApply
+
+
+class ZeroOut(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Zero out all entries, for testing purposes."""
+
+    def __init__(self):
+        super(ZeroOut, self).__init__()
+
+    def augment_image(self, image, transformation=None):
+        return 0 * image
+
+    def augment_label(self, label, transformation=None):
+        return 0 * label
+
+    def augment_bounding_box(self, bounding_box, transformation=None):
+        return 0 * bounding_box
+
+
+class MaybeApplyTest(tf.test.TestCase, parameterized.TestCase):
+    rng = tf.random.Generator.from_non_deterministic_state()
+
+    @parameterized.parameters([-0.5, 1.7])
+    def test_raises_error_on_invalid_rate_parameter(self, invalid_rate):
+        with self.assertRaises(ValueError):
+            MaybeApply(rate=invalid_rate, layer=ZeroOut())
+
+    def test_works_with_batched_input(self):
```

Comment:
can you pass a seed so that this test is not potentially flaky?  Given, it is 1/2^32 flakiness, but still may as well seed it.

Reply:
Added seed to `rng` on line 37.

Reply:
Thanks!  does this seed the layer too?

Reply:
You are right. Added `seed` param to layer as well.